test/unit/test.xpi.js

```diff
@@ -45,13 +46,29 @@ describe("lib/xpi", function() {
     })
     .catch(done);
   });
+  it("Check file name with id", function(done) {
+    process.chdir(simpleAddonWithIdPath);
+    var manifest = require(path.join(simpleAddonWithIdPath, "package.json"));
+    var xpiPath;
+    return xpi(manifest).then(function(filePath) {
+      xpiPath = filePath;
+      expect(xpiPath).to.be.equal(path.join(simpleAddonWithIdPath,
+                                            "@simple-addon-1.0.0.xpi"));
+      return true
```

Comment:
you could do `return xpiPath` and `.then(function(xpiPath) {...})` on the following line to simplify this.